dashboard/src/main/java/com/google/cloud/tools/opensource/dashboard/DashboardArguments.java

```diff
@@ -97,13 +100,34 @@ static DashboardArguments readCommandLine(String... arguments) throws ParseExcep
 
     try {
       // Throws ParseException if required option group ('-f' or '-c') is not specified
-      return new DashboardArguments(parser.parse(options, arguments));
+      CommandLine commandLine = parser.parse(options, arguments);
+      String dependencyMediationValue = commandLine.getOptionValue('m');
+      if (dependencyMediationValue != null && !allowsDependencyMediationValues.contains(dependencyMediationValue)) {
+        throw new ParseException("Valid values for '-m' are " + allowsDependencyMediationValues);
+      }
+
+      return new DashboardArguments(commandLine);
     } catch (ParseException ex) {
       helpFormatter.printHelp("DashboardMain", options);
       throw ex;
     }
   }
 
+  enum DependencyMediationAlgorithm {
+    MAVEN, GRADLE,
+  }
+
+  /**
+   * Returns dependency mediation algorithm. By default it's {@link DependencyMediationAlgorithm#MAVEN}.
+   */
+  DependencyMediationAlgorithm getDependencyMediation() {
```

Comment:
Because I want GradleDependencyMediation to take a BOM for enforced platform, this method does not return `DependencyMediation`. (This method does not know the content of the BOM.)